### PR TITLE
Add some useful Rmd snippets

### PR DESCRIPTION
**Pull request by jooyoungseo**
# What problem did you solve?

Closing #1008

## (If you do not have screenshot) How can I check this pull request?
1. Open Rmd file.

1. Insert the following snippets:

  - [x] inline r code

  - [x] inline footnotes

  - [x] speaker notes

  - [x] two columns

  - [x] reference list

  - [x] code chunk

  - [x] incremental list

  - [x] nonincremental list

  - [x] incremental fenced block

  - [x] nonincremental fenced block

  - [x] unordered item

  - [x] ordered item

  - [x] task list

  - [x] checked item

  - [x] unchecked item

  - [x] strikeout

  - [x] superscript

  - [x] subscript

  - [x] bookdown theorem

  - [x] cross-reference bookdown theorem

  - [x] bookdown lemma

  - [x] cross-reference bookdown lemma

  - [x] bookdown corollary

  - [x] cross-reference bookdown corollary

  - [x] bookdown proposition

  - [x] cross-reference bookdown proposition

  - [x] bookdown conjecture

  - [x] cross-reference bookdown conjecture

  - [x] bookdown definition

  - [x] cross-reference bookdown definition

  - [x] bookdown example

  - [x] cross-reference bookdown example

  - [x] bookdown exercise

  - [x] cross-reference bookdown exercise

  - [x] bookdown hypothesis

  - [x] cross-reference bookdown hypothesis

  - [x] bookdown equation

  - [x] cross-reference bookdown equation

  - [x] figure chunk

  - [x] external figure chunk

  - [x] cross-reference bookdown figure

  - [x] cross-reference bookdown table

  - [x] bookdown section label

  - [x] cross-reference bookdown section

  - [x] bookdown part

  - [x] bookdown unnumbered part

  - [x] bookdown appendix

### CODE REVIEW

**Comment by jooyoungseo** [comment]
Also added all the supported languages to Insert code chunk snippet. Users now can  interactively choose the supported languages.

The following languages are currently available:

``` r
names(knitr::knit_engines$get())
#>  [1] "awk"       "bash"      "coffee"    "gawk"      "groovy"    "haskell"  
#>  [7] "lein"      "mysql"     "node"      "octave"    "perl"      "psql"     
#> [13] "Rscript"   "ruby"      "sas"       "scala"     "sed"       "sh"       
#> [19] "stata"     "zsh"       "asis"      "asy"       "block"     "block2"   
#> [25] "bslib"     "c"         "cat"       "cc"        "comment"   "css"      
#> [31] "dot"       "embed"     "fortran"   "fortran95" "go"        "highlight"
#> [37] "js"        "julia"     "python"    "R"         "Rcpp"      "sass"     
#> [43] "scss"      "sql"       "stan"      "targets"   "tikz"      "verbatim" 
#> [49] "glue"      "glue_sql"  "gluesql"
```

<sup>Created on 2022-02-21 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

`stan` and `sql` snippets have followed the recommended templates by [R Markdown: The Definitive Guide](https://bookdown.org/yihui/rmarkdown/language-engines.html).

**Comment by renkun-ken** [comment]
We are about to release a new version. Would you like to include this PR in the upcoming release?

**Comment by jooyoungseo** [comment]
I will have more time tomorrow to finish this PR if that works for you. I don't mind if you merge this after the new release for the next round.

**Review by renkun-ken — APPROVED** [review]
LGTM Thanks!

File: snippets/rmarkdown.json [hunk 1]
```diff
@@ -71,7 +116,7 @@
 	"Insert code chunk": {
 		"prefix": "code chunk",
 		"body": [
-			"```{${1:language}}",
+			"```{${1|language,awk,bash,coffee,gawk,groovy,haskell,lein,mysql,node,octave,perl,psql,Rscript,ruby,sas,scala,sed,sh,stata,zsh,asis,asy,block,block2,bslib,c,cat,cc,comment,css,dot,embed,fortran,fortran95,go,highlight,js,julia,python,R,Rcpp,sass,scss,sql\\, connection=db,stan\\, output.var=\"stanmodel\",targets,tikz,verbatim|}}",
```
Comment (renkun-ken) [inline snippets/rmarkdown.json:119]: I'm trying this, but I don't manage to see these languages anywhere in the completion list when I insert this snippet. Am I missing something?

Comment (renkun-ken) [reply snippets/rmarkdown.json:119]: I'm thinking of adding this to the completion provider for R markdown.

Comment (renkun-ken) [reply snippets/rmarkdown.json]: I revert this snippet for the moment for future revist.

Comment (renkun-ken) [reply snippets/rmarkdown.json]: Looks like it is related to https://github.com/microsoft/vscode/issues/144362.

Comment (jooyoungseo) [reply snippets/rmarkdown.json]: Yes, it stopped working after the latest VSCode release. :(

**Pull request merged**